src/lib.rs

```diff
@@ -24,6 +24,20 @@ pub enum Error<E> {
 /// I²C address
 pub const ADDRESS: u8 = 0x40;
 
+pub struct MilliCelsius(i32);
+pub struct RHPerCentMille(i32);
```

Comment:
I'd simply call those types `Temperature` and `RelativeHumidity`. The meaning of the internal number is already conveyed by the method names.

Reply:
Hmm.  I followed the Rust styleguide, which has a closely analogous example, and names the types according to the units used:

https://doc.rust-lang.org/1.0.0/style/features/types/newtype.html

I suppose I feel that:
```rust
fn convert_raw_temperature(temp_raw: u16) -> MilliCelsius {
```
is clearer than:
```rust
fn convert_raw_temperature(temp_raw: u16) -> Temperature {
```
... especially as the formula quoted immediately above from the datasheet calculates its answer in Celsius, but the fixed point implementation in the function itself calculates milli-Celsius instead.

Any thoughts?

Reply:
The internal implementation does not really matter, as long as it represents a temperature and as long as it's clear what the values that are returned by the methods mean.